modules/kn-service-apply.adoc

```diff
@@ -0,0 +1,37 @@
+// Module included in the following assemblies:
+//
+// * serverless/knative_serving/serverless-applications.adoc
+// * serverless/cli_reference/kn-services-ref.adoc
+
+[id="kn-service-apply_{context}"]
+= Applying service declarations
+
+You can apply declarations to a Knative service by using the `kn` CLI `kn service apply` command.
+
```

Comment:
Should we add some extra explanation here and explain e.g. the difference between `kn service apply` and `kn service update` ? I can deliver some input here.

Reply:
If we're going to add it we should maybe add it to both the apply and update topics since these are separate docs modules and we shouldn't assume folks will always read both. If you can provide some input about what you'd like to say I'd be happy to add it.

Reply:
I would write something along the lines:

----

You can declaratively configure a Knative service by using the `kn service apply` command. 
If the service does not exist it will be created, otherwise, the existing service will be updated with the options that have been changed.
When using `kn service apply` you have to provide the full configuration of the Knative service. 
This is different to the `kn service update` command which only requires the options that need to be updated.

----

Please feel free to reword to fit the required conventions and/or style.
I could add more info (like why and in which context `kn service apply` is useful) but afair that is not what you are looking for this reference documentation.

Reply:
Maybe also mention that `kn service apply` is _idempotent_ , i.e. if you call it a second time with the same options then nothing changes.

Reply:
> I could add more info (like why and in which context kn service apply is useful) but afair that is not what you are looking for this reference documentation.

If you feel like this is something you'd have information for I'd be happy to have the input, it might not do any harm to call out examples of why someone would use this 🙂

Reply:
`kn service apply` is especially useful when used in an automated way, like in a shell script or a  continuous integration pipeline. In this context, you typically want to fully specify the state of the service in a single command to _declare_ the target state. In contrast `kn service update` is more suitable for an interactive session on the command line where a user builds up the service incrementally, step by step (and eventually export it at the end as YAML file to check-in into an SCM), as for `kn service updae` you only have to specify the changes that you want to update.

^^ these are the typical use cased for both commands, respectively.